File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        
        pre = strs[0]
        
        for string in strs[1:]:
            while string.find(pre) != 0:
                pre = pre[:-1]
                if pre == "":
                    return ""
        
        return pre
        

File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()){
            return "";
        }
        
        std::string prefix = strs[0];
        for (int j = 1; j < strs.size(); j++) {
            std::string current = strs[j];
            int i = 0;
            while (i < prefix.length() && i < current.length() && prefix[i] == current[i]) {
                i++;
            }
            prefix = prefix.substr(0, i);
            if (prefix.empty()){
                return "";
            }
        }
        return prefix;
    }
};
